apps/backend/src/controllers/neighborhoods.ts

```diff
@@ -105,4 +107,102 @@ neighborhoodsRouter.post('/:id/join', middleware.userIdExtractorAndLoginValidato
   return res.status(201).send({ success: 'You have joined the neighborhood' });
 }));
 
+// Get a neighborhood's requests
+neighborhoodsRouter.get('/:id/requests', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const neighborhoodID = Number(req.params.id);
+  // LoginValidator ensures that loggedUserId is present
+  const loggedUserID = req.loggedUserId as number;
+
+  const isUserMemberOfNeighborhood: boolean = await neighborhoodServices
+    .isUserMemberOfNeighborhood(loggedUserID, neighborhoodID);
+
+  if (!isUserMemberOfNeighborhood) {
+    return res.status(400).send({ error: 'user is not a member of the neighborhood' });
+  }
+
+  const requests: RequestData[] = await neighborhoodServices
+    .getRequestsAssociatedWithNeighborhood(neighborhoodID);
+
+  return res.status(200).send(requests);
+}));
+
+// Get a single request
+neighborhoodsRouter.get('/:neighborhoodId/requests/:requestId', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const requestId = Number(req.params.requestId);
+  const neighborhoodId = Number(req.params.neighborhoodId);
+  const loggedUserId = req.loggedUserId as number;
+
+  const isUserMemberOfNeighborhood = await neighborhoodServices
+    .isUserMemberOfNeighborhood(loggedUserId, neighborhoodId);
+
+  if (!isUserMemberOfNeighborhood) {
+    return res.status(401).send({ error: 'user not a member of neighborhood' });
+  }
+
+  const isRequestAssociatedWithNeighborhood = await neighborhoodServices
+    .isRequestAssociatedWithNeighborhood(requestId, neighborhoodId);
+
+  if (!isRequestAssociatedWithNeighborhood) {
+    return res.status(400).send({ error: 'request not associated with the neighborhood' });
+  }
+
+  const request: RequestData = await requestServices.getRequestById(requestId);
+  return res.status(200).send(request);
+}));
+
+// Create request
+neighborhoodsRouter.post('/:neighborhoodId/requests', middleware.userIdExtractorAndLoginValidator, catchError(async (req: RequestWithAuthentication, res: Response) => {
+  const postData: CreateRequestData = await requestServices.parseCreateRequestData(req.body);
+
+  const loggedUserId: number = req.loggedUserId as number;
+  const neighborhoodId: number = Number(req.params.neighborhoodId) as number;
+
+  const request: RequestData = await neighborhoodServices
+    .createRequest(postData, loggedUserId, neighborhoodId);
+
+  return res.status(201).send(request);
+}));
+
+// Update request
+neighborhoodsRouter.put(
```

Comment:
Since update route requires only requestId, we can move it to request.ts.

Reply:
You're probably right. We should make sure that we won't need to associate a request with a neighborhood for all these routes before making the change, though. We don't want to keep moving them from one place to another 😪

Reply:
The request is associated with its neighborhood when created, and will be part of the record as a foreign key, yes? That key can't be altered or deleted except when deleting the entire record, so it should be available in any context once the request is created, right?

Reply:
I think it's safe to move it to request.ts